### PR TITLE
[Jonathan Iheme] - Sage - How to Build a Full Stack Social Media Dapp on the Celo Blockchain

### PR DESCRIPTION
**Pull request by 4undRaiser**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+
+The next function is the `sendTip()`. This function will be used to send a tip in cUSD to a specific user from a list of posts. we first added checks to make sure the transfer of funds is possible using the `IERC20Token` contract. It then uses the `transferFrom` function, which allows a third party to transfer funds on behalf of the sender, to send the tip to the specified user. It takes three parameters, the `sender`, the `user` to receive the tip, and the amount of the tip. If the transfer is successful, the code returns a success message, otherwise, it will return an error.
+
+```solidity
+    function getPostsLength() public view returns(uint){
+        return(postLength);
+    }   
+```
+
+Finally, the `getPostsLength()` function allows users to view the current length of the `posts array`. This function takes no arguments and returns the value of the postLength variable.
```
Comment (Ishanpathak1) [inline blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md:203]: Can you add,whole code like --> Your whole code should look like this

File: blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        addComment={addComment}
+        sendTip={sendTip}
+        walletAddress={address}
+      />
+    </div>
+  );
+
+export default App;
+```
+And finally, we render the App component and return the Home and Posts components with the necessary props.
```
Comment (Ishanpathak1) [inline blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md:404]: Here also, can you just add the whole code--> this should look like this

File: blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
+---
+title: How to Build a Full Stack Social Media Dapp on the Celo Blockchain
+description: This tutorial covers how to build a decentralized social media dapp on the celo blockchain.
+authors:
+  - name: ✍️ Jonathan Iheme
+url: https://github.com/4undRaiser
+image_url: https://avatars.githubusercontent.com/u/87926451?s=96&v=4
+tags: [celosage, solidity, celo]
+hide_table_of_contents: false
```
Comment (Ishanpathak1) [inline blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md:9]: it has to be set to true for now

File: blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md [hunk 4]
```diff
@@ -0,0 +1,418 @@
+---
+title: How to Build a Full Stack Social Media Dapp on the Celo Blockchain
+description: This tutorial covers how to build a decentralized social media dapp on the celo blockchain.
+authors:
+  - name: ✍️ Jonathan Iheme
+url: https://github.com/4undRaiser
+image_url: https://avatars.githubusercontent.com/u/87926451?s=96&v=4
+tags: [celosage, solidity, celo]
```
Comment (Ishanpathak1) [inline blog/02-10-23-How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain/How-to-Build-a-Full-Stack-Social-Media-Dapp-on-the-Celo-Blockchain.md:8]: You can add react also in a tags

**Review by Ishanpathak1 — COMMENTED** [review]
It would be great for the articles if you would make the changes suggested by me

**Comment by 4undRaiser** [comment]
Done

**Comment by viral-sangani** [comment]
@4undRaiser Mark the reviews by clicking on `Resolve Conversation` once you are done making those changes.

**Comment by viral-sangani** [comment]
Build is failing here so cannot merge this.
@Ishanpathak1 @4undRaiser the url slug for the article in smallcase. Please change that.

**Comment by 4undRaiser** [comment]
@Ishanpathak1 @viral-sangani I have fixed the problem. you can merge now.

**Review by viral-sangani — APPROVED** [review]

**Comment by viral-sangani** [comment]
LGTM

**Pull request merged**